### PR TITLE
NEW Sepa mandate online Signature

### LINKED ISSUE #24637: Add signatures to SEPA direct debit authorizations
### Feature Request

- [ ] Add signature on each SEPA
- [ ] Add Signature on each line
- [ ] Add statuts on line

image

### Use case

_No response_

### Suggested implementation

_No response_

### Suggested steps

_No response_

### PR DESCRIPTION
**Pull request by FHenry**
fix: #24637 

https://github.com/Dolibarr/dolibarr/assets/1050053/2269d177-b31c-46dc-9cd5-a132bbc176b7

### CODE REVIEW

File: htdocs/admin/bank.php [hunk 1]
```diff
@@ -415,7 +415,7 @@
 print $langs->trans('BankColorizeMovementDesc');
 print "</td>";
 // Active
-if ($conf->global->BANK_COLORIZE_MOVEMENT) {
+if (!empty(getDolGlobalInt('BANK_COLORIZE_MOVEMENT'))) {
```
Comment (eldy) [inline htdocs/admin/bank.php:418]: empty is not allowed with getDolGlobalInt.
May be you want to use the !

File: htdocs/admin/bank.php [hunk 2]
```diff
@@ -482,8 +482,18 @@
 	print '<a class="reposition" href="'.$_SERVER["PHP_SELF"].'?action=setreportlastnumreleve&token='.newToken().'">'.img_picto($langs->trans("Disabled"), 'switch_off').'</a>';
 	print "</td>";
 }
-
 print "</tr>\n";
+
+// Allow SEPA Mandate OnLine Sign
+if (empty(getDolGlobalInt('SOCIETE_DISABLE_BANKACCOUNT'))) {
```
Comment (eldy) [inline htdocs/admin/bank.php:488]: Same here. May be you want to do 
!getDolGlobal...

**Comment by FHenry** [comment]
@eldy all test are ok, can you check ?

File: htdocs/core/lib/signature.lib.php [hunk 3]
```diff
@@ -146,6 +146,21 @@ function getOnlineSignatureUrl($mode, $type, $ref = '', $localorexternal = 1)
 		} else {
 			$out .= '&securekey='.dol_hash($securekeyseed.$type.$ref.(isModEnabled('multicompany') ? (is_null($object) ? '' : $object->entity) : ''), '0');
 		}
+	} else {
+		$securekeyseed = getDolUserString(dol_strtoupper($type).'ONLINE_SIGNATURE_SECURITY_TOKEN');
```
Comment (eldy) [inline htdocs/core/lib/signature.lib.php:150]: Are you sure about getDolUserString ?
Not getDolGlobalString ?

Comment (FHenry) [reply htdocs/core/lib/signature.lib.php]: @eldy, yes don't know why, autocompletion of IDE put this first on this file, has it work I didn't see it. I've also change all other $conf->global in this page

**Comment by FHenry** [comment]
@eldy you've set tag discussions, does it mean you want to have more information about this feature. 
I've put a video to demonstrate ans all CI treat are ok, it if it can help to merge.

**Pull request merged**